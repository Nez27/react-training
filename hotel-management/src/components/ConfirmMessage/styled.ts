import styled from "styled-components";

const StyledConfirmDelete = styled.div`
  width: 450px;

  & p {
    font-size: var(--fs-sm);
    text-align: center;
    margin-bottom: 30px;
  }

  & div {
    display: flex;
    justify-content: space-around;

    padding-inline: 100px;
  }
`;

export { StyledConfirmDelete };
