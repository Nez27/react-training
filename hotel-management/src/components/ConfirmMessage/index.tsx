import { memo, useCallback } from 'react';

// Styled
import { StyledConfirmDelete } from './styled';

// Component
import Button from '@commonStyle/Button';

interface IConfirmMessage {
  message: string;
  onConfirm: () => void;
  onCloseModal?: () => void;
}

const ConfirmMessage = memo(
  ({ message, onConfirm, onCloseModal }: IConfirmMessage) => {
    const handleConfirmBtn = useCallback(() => {
      onConfirm();
      onCloseModal!();
    }, [onCloseModal, onConfirm]);

    return (
      <StyledConfirmDelete>
        <p>{message}</p>

        <div>
          <Button variations="danger" onClick={handleConfirmBtn}>
            Yes
          </Button>
          <Button variations="secondary" onClick={onCloseModal}>
            No
          </Button>
        </div>
      </StyledConfirmDelete>
    );
  }
);

export default ConfirmMessage;
