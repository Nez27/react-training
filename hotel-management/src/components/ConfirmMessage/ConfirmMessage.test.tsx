import renderer from 'react-test-renderer';

// Components
import ConfirmMessage from '.';

describe('ConfirmMessage', () => {
  const isDisable = true;
  const message = 'Hello World!';
  const handleOnConfirm = jest.fn();

  const wrapper = renderer.create(
    <ConfirmMessage
      disabled={isDisable}
      message={message}
      onConfirm={handleOnConfirm}
    />
  );

  test('Should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
