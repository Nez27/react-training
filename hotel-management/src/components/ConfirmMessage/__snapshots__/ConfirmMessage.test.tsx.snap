// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmMessage Should render correctly 1`] = `
<div
  className="sc-bdnyFh dukbvo"
>
  <p>
    Hello World!
  </p>
  <div>
    <button
      className="sc-gtsqUy eZzwwl"
      disabled={true}
      onClick={[MockFunction]}
      variations="danger"
    >
      Delete
    </button>
    <button
      className="sc-gtsqUy JDRgw"
      disabled={true}
      variations="secondary"
    >
      Cancel
    </button>
  </div>
</div>
`;
